Unused static int pushresult() on Windows

When using more strict checking (e.g. `-Wall` or `-Wunused-function`) on platforms where `_WIN32` is defined, the compiler may complain:
```
lfs.c:152:12: warning: ‘pushresult’ defined but not used [-Wunused-function]
 static int pushresult(lua_State *L, int i, const char *info)
            ^
```
This is due to the fact that `make_link()` won't use `pushresult()` and will always fail with `pusherror()` instead.

It might be desirable to either
a) encapsulate the `pushresult` declaration with a `#ifndef _WIN32` / `#endif` block, or
b) change the `pusherror(L, "make_link is not supported on Windows");` to `pushresult(L, -1, "make_link is not supported on Windows");` instead.
(I actually prefer the latter.)

[EDIT] An assignment like `errno = ENOSYS;` could also be a good idea, so the subsequent `strerror(errno)` returns something meaningful.

Regards, NiteHawk

--- a/src/lfs.c
+++ b/src/lfs.c
@@ -431,7 +431,8 @@ static int make_link(lua_State *L)
         return pushresult(L,
                 (lua_toboolean(L,3) ? symlink : link)(oldpath, newpath), NULL);
 #else
-        return pusherror(L, "make_link is not supported on Windows");
+        errno = ENOSYS; /* = "Function not implemented" */
+        return pushresult(L, -1, "make_link is not supported on Windows");
 #endif
 }
 

